packages/component_code_gen/helpers/embeddings_similarity_search.py

```diff
@@ -0,0 +1,62 @@
+from config.config import config
+from sklearn.metrics.pairwise import cosine_similarity
+import numpy as np
+import requests
+
+
+def get_embedding(text):
+    if config["openai_api_type"] == "azure":
+        url = f"{config['azure']['api_base']}/openai/deployments/{config['azure']['deployment_name']}/embeddings"
+        api_key = config["azure"]["api_key"]
+    else:
+        url = "https://api.openai.com/v1/embeddings"
+        api_key = config["openai"]["api_key"]
+
+    headers = {
+        "Authorization": f"Bearer {api_key}",
+    }
+
+    data = {
+        "input": text,
+        "model": "text-embedding-ada-002",
+    }
+
+    response = requests.post(url, headers=headers, json=data)
+
+    if not response.ok:
+        raise Exception(
+            f"OpenAI exception: {response.status_code} - {response.text}")
+
+    embedding = response.json()["data"][0]["embedding"]
+    return np.asarray(embedding).reshape(1, -1)
+
+
+def split_document(document, chunk_size=1000):
```

Comment:
You might try using https://python.langchain.com/docs/modules/data_connection/document_transformers/text_splitters/HTML_header_metadata or thinking about how to split the documents into logical sections vs. just chunking on size. This is fine for now and I would merge if you're seeing perf improvements, but chunking is so critical that we can iterate on it in the future and see if we see further increases.